File: the_data_mine/projects/project01/jason-ivey-project01.R
#------------------------Q2------------------------
# **What is the name of the node on Anvil that you are running on?**
system("hostname", intern=TRUE)


#------------------------Q5------------------------
# From: https://stackoverflow.com/questions/13023274/how-to-do-printf-in-r
printf <- function(...) invisible(print(sprintf(...)))

cores_a = 1000 * 128
cores_b = 32 * 128
cores_c = 16 * 128

mem_a =  1000*256
mem_b = 32*1
mem_c = 16*512

printf(c("A: total cores ==%.2f", ", total memory =%.2f"), c(cores_a, mem_a))
printf(c("B: total cores ==%.2f", ", total memory =%.2f"), c(cores_b, mem_b))
printf(c("C: total cores ==%.2f", ", total memory =%.2f"), c(cores_c, mem_c))


#------------------------Q6------------------------
dat <- read.csv("/anvil/projects/tdm/data/disney/flight_of_passage.csv")

head(dat)

File: the_data_mine/projects/project01/jason-ivey-project01.py
#------------------------Q3------------------------
# Practice running the following examples.

my_list = [1, 2, 3]
print(f'My list is: {my_list}')

#------------------------Q5------------------------

# In question (1) we answered questions about cores
# and memory for the Anvil clusters. To do so, we 
# needed to perform some arithmetic. Instead of 
# using a calculator (or paper, or mental math for 
# you good-at-mental-math folks), write these 
# calculations using R and Python, in separate code 
# cells.

(cores_a, mem_a) = ( 1000 * 128, 1000*256)
(cores_b, mem_b) = ( 32 * 128, 32*1)
(cores_c, mem_c) = ( 16 * 128, 16*512)

print(f"""
A: total cores = {cores_a}, total memory = {mem_a} GB
B: total cores = {cores_b}, total memory = {mem_b} TB 
C: total cores = {cores_c}, total memory = {mem_c} GB 
""")